### PR TITLE
Flatpak manifest for org.flatpak.galacteek.Galacteek

### PR DESCRIPTION
**Pull request by pinnaculum**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. Needed permissions: pulseaudio (galacteek has a mediaplayer), network (it uses IPFS), and all devices (for access to cameras). I am not sure if *filesystem=home* is needed.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by pinnaculum** [comment]
bot, build org.flatpak.galacteek.Galacteek

**Comment by pinnaculum** [comment]
bot, build org.flatpak.galacteek.Galacteek

**Comment by pinnaculum** [comment]
Seems like *build-args* isn't allowed by flatpak-builder in the buildbot environment .. Will read more on this.

File: org.flatpak.galacteek.Galacteek.yaml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+      - pip install -r requirements.txt
+      - pip install -r requirements-extra-markdown.txt
+      - pip install -r requirements-ui-pyqt-5.15.txt
+      - python3 setup.py build build_docs install
+    build-options:
+      build-args:
+        - --share=network
+    sources:
+      - type: git
+        url: https://github.com/pinnaculum/galacteek
```
Comment (nedrichards) [inline org.flatpak.galacteek.Galacteek.yaml:128]: this is going to need to be a specific release, commit or tag.

File: org.flatpak.galacteek.Galacteek.yaml [hunk 2]
```diff
@@ -0,0 +1,167 @@
+app-id: org.flatpak.galacteek.Galacteek
```
Comment (nedrichards) [inline org.flatpak.galacteek.Galacteek.yaml:1]: based on the location of the source code this should be something like `com.github.pinnaculum.galacteek` or `io.github.pinnaculum.galacteek` (either is fine, using flatpak.org is not)

**Comment by nedrichards** [comment]
Builds args is allowed but network access is not. You need to specify all your dependencies within the manifest for reproducibility. See https://github.com/flathub/flathub/wiki/App-Requirements#stable-releases-reproducible-builds for more.

**Comment by pinnaculum** [comment]
@nedrichards Thanks a lot for your help on this.

Given that it builds well outside of the flathub buildbot environment, i'm thinking of releasing it as a single-file flatpak bundle with the gh release. Are there any other flatpaks that use miniconda ?

**Comment by pinnaculum** [comment]
Closing

**Comment by nedrichards** [comment]
@pinnaculum sure! go for it. We'd always recommend that you try and host a flatpak repo rather than just the bundle if you can https://docs.flatpak.org/en/latest/hosting-a-repository.html that way users can get updates as you release them and the whole workflow is a lot cleaner. Your choice though!

**Pull request closed without merging**